src/asset-swapper/utils/market_operation_utils/orders.ts

```diff
@@ -564,9 +568,25 @@ export function createNativeOptimizedOrder(
         fillData,
         fill: cleanFillForExport(fill),
     };
-    return fill.type === FillQuoteTransformerOrderType.Rfq
-        ? { ...base, type: FillQuoteTransformerOrderType.Rfq, fillData: fillData as NativeRfqOrderFillData }
-        : { ...base, type: FillQuoteTransformerOrderType.Limit, fillData: fillData as NativeLimitOrderFillData };
+    switch (fill.type) {
+        case FillQuoteTransformerOrderType.Rfq:
+            return { ...base, type: FillQuoteTransformerOrderType.Rfq, fillData: fillData as NativeRfqOrderFillData };
+        case FillQuoteTransformerOrderType.Limit:
+            return {
+                ...base,
+                type: FillQuoteTransformerOrderType.Limit,
+                fillData: fillData as NativeLimitOrderFillData,
+            };
+        case FillQuoteTransformerOrderType.Otc:
+            return { ...base, type: FillQuoteTransformerOrderType.Otc, fillData: fillData as NativeOtcOrderFillData };
+        case FillQuoteTransformerOrderType.Bridge:
+            throw new Error('BridgeOrder is not a Native Order');
+
+        default:
+            ((_x: never) => {
```

Comment:
doing this such that TS will complain and not compile if a new order type is added. More context on this pattern: https://www.carlrippon.com/make-switch-statements-super-type-safe-with-the-never-type/

it's something we use on the RFQ side of things to ensure our switch statements throw compile time complaints (rather than us finding much later that the default case got hit)